Userland/Libraries/LibWeb/Fetch/Infrastructure/FetchParams.h

```diff
@@ -0,0 +1,99 @@
+
+    [[nodiscard]] NonnullRefPtr<Request> request() const { return m_request; }
+    [[nodiscard]] NonnullRefPtr<FetchController> controller() const { return m_controller; }
+    [[nodiscard]] NonnullRefPtr<FetchTimingInfo> timing_info() const { return m_timing_info; }
+
+    [[nodiscard]] NonnullRefPtr<FetchAlgorithms> algorithms() const { return m_algorithms; }
+    void set_algorithms(NonnullRefPtr<FetchAlgorithms> algorithms) { m_algorithms = move(algorithms); }
+
+    [[nodiscard]] TaskDestination& task_destination() { return m_task_destination; }
+    [[nodiscard]] TaskDestination const& task_destination() const { return m_task_destination; }
+    void set_task_destination(TaskDestination task_destination) { m_task_destination = move(task_destination); }
+
+    [[nodiscard]] HTML::CanUseCrossOriginIsolatedAPIs cross_origin_isolated_capability() const { return m_cross_origin_isolated_capability; }
+    void set_cross_origin_isolated_capability(HTML::CanUseCrossOriginIsolatedAPIs cross_origin_isolated_capability) { m_cross_origin_isolated_capability = cross_origin_isolated_capability; }
+
+    [[nodiscard]] PreloadedResponseCandidate& preloaded_response_candidate() { return m_preloaded_response_candidate; }
+    [[nodiscard]] PreloadedResponseCandidate const& preloaded_response_candidate() const { return m_preloaded_response_candidate; }
+    void set_preloaded_response_candidate(PreloadedResponseCandidate preloaded_response_candidate) { m_preloaded_response_candidate = move(preloaded_response_candidate); }
+
+    [[nodiscard]] bool is_aborted() const;
+    [[nodiscard]] bool is_canceled() const;
+
+private:
+    FetchParams(NonnullRefPtr<Request>, NonnullRefPtr<FetchTimingInfo>);
+
+    // https://fetch.spec.whatwg.org/#fetch-params-request
+    // request
+    //     A request.
+    NonnullRefPtr<Request> m_request;
+
+    // https://fetch.spec.whatwg.org/#fetch-params-process-request-body
+    // process request body chunk length (default null)
+    // https://fetch.spec.whatwg.org/#fetch-params-process-request-end-of-body
+    // process request end-of-body (default null)
+    // https://fetch.spec.whatwg.org/#fetch-params-process-early-hints-response
+    // process early hints response (default null)
+    // https://fetch.spec.whatwg.org/#fetch-params-process-response
+    // process response (default null)
+    // https://fetch.spec.whatwg.org/#fetch-params-process-response-end-of-body
+    // process response end-of-body (default null)
+    // https://fetch.spec.whatwg.org/#fetch-params-process-response-consume-body
+    // process response consume body (default null)
+    //     Null or an algorithm.
+    NonnullRefPtr<FetchAlgorithms> m_algorithms;
+
+    // https://fetch.spec.whatwg.org/#fetch-params-task-destination
+    // task destination (default null)
+    //     Null, a global object, or a parallel queue.
+    TaskDestination m_task_destination;
```

Comment:
Who protects this from garbage collection if it's a `JS::NonnullGCPtr<JS::Object>`?